### ISSUE TITLE
Force Close Forground App Action

### ISSUE THREAD

**Issue by Friday13th87** [toneiv/UbikiTouch#100]
Is there any way to get a function / action to force close the current foreground app?
i tried to record an action by recording while killing a foreground app with another application, but it was not recording the process actually.
Is there a way to write a *.bin file for the action and then import it?
the command i was thinking about would be:
`package=`adb shell dumpsys window windows | grep mCurrentFocus | grep -oP '(?<=\s).\S*?(?=/)'` 
adb shell am force-stop $package`

**Comment by toneiv** [comment]
I have done some tests and I think I can come up with a solution. However, unless I am mistaken, this is an action that requires root access.
Is this your case or have you considered another solution?

**Comment by Friday13th87** [comment]
No actually the ADB command does not require root.
Its looking up the foreground app, saving it and send a force close. 
But i think i made a mistake in the first command for the lookup, will have a look at it on the weekend.

You can try it by just connect to a adb shell and type "am force-close anypackage" for example com.whatsapp - no root needed

I tought maybe there is a way to record the action going the system settings way, like
--> opening app info
--> pressing/sending command for force close
as this action-recording is already implemented

**Comment by toneiv** [comment]
Yes, but I don't think it is possible to execute an ADB command to the device from the device itself. 
That's why I think root is necessary. 

For information I can perform this action as root (so it can be transposed to adb)
- **dumpsys activity recents | grep "Recent #0"** returns the application package
- **am force-stop mypackage**

The second solution implies that opening the info screen and pressing the button will be perceived by the user, so I'm not too sure that's a good idea. I'll keep digging though.

**Comment by Friday13th87** [comment]
you can use for example "termux" and use the command with it, so its working from the device itself, but i dont know if its usable by sending a single command by each individual application or if you need something more.

**Comment by toneiv** [comment]
I haven't had much time to dig into it but I think Termux is more of an emulator and doesn't really give access to the device. 
It don't have elevated privileges as ADB does. No am force-stop function available and impossible to list the running applications for example.

I'll still look into it and see if I can add a root function that meets this need.
